packages/server/modules/emails/clients/jsonEcho.ts

```diff
@@ -0,0 +1,53 @@
+import { createTransport } from 'nodemailer'
+import type { EmailTransport } from '@/modules/emails/domain/types'
+import type { Logger } from '@/observability/logging'
+import { EmailTransportInitializationError } from '@/modules/emails/errors'
+import type { SentMessageInfo } from 'nodemailer/lib/json-transport'
+import { SentEmailDeliveryStatus } from '@/modules/emails/domain/consts'
+
+const createJsonEchoTransporter = () => {
+  const newTransport = createTransport({ jsonTransport: true })
+  const wrappedTransporter: EmailTransport = {
```

Comment:
we should provide an internal storage mechanism here, so that we can make assertions on send emails